### PR TITLE
Template 5 fixes

### PR DESCRIPTION
**Pull request by maxkonovalov**
# What has changed

A separate copy of the `Template5View` -> `LinTemplate5View` has been created, and all the changes have been applied to the new template modification. The `template5` option is now pointing to the custom implementation by default.

The paywall layout has been updated to fit the most important content whenever possible.

# How it was changed

All UI elements are sized to fit the contents without trimming or scrolling, whenever possible. The rest of the space is filled with the image at the top. There is an exception to this for an iPhone in a landscape orientation. In that case, the image is hidden, and the whole content becomes scrollable.

The top image is added as a background of a Spacer. It allows cropping the image based on available space while keeping its original aspect ratio. At the same time, this approach does not affect the rest of the layout.

The views have been moved from the `scrollableContent` to the `content` property, and the `content` itself is made `scrollableIfNecessary`.

Other minor changes have been added to the spacings and view frames.

A new constant, `defaultContentWidth` has been added. The main content is constrained to this size horizontally.

# Example

| iPad Portrait | iPad Landscape | iPhone Portrait | iPhone Landscape |
| --- | --- | --- | --- |
| Simulator Screenshot - iPad Pro (11-inch) (4th generation) - 2023-12-28 at 22 39 04 | Simulator Screenshot - iPad Pro (11-inch) (4th generation) - 2023-12-28 at 22 38 23 | Simulator Screenshot - iPhone 15 Pro - 2023-12-28 at 22 44 55 | Simulator Screenshot - iPhone 15 Pro - 2023-12-28 at 22 45 04 |

### CODE REVIEW

File: RevenueCatUI/Templates/LinTemplate5View.swift [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+    init(_ configuration: TemplateViewConfiguration) {
+        self._selectedPackage = .init(initialValue: configuration.packages.default)
+        self.configuration = configuration
+        self._displayingAllPlans = .init(initialValue: configuration.mode.displayAllPlansByDefault)
+    }
+
+    var body: some View {
+        self.content
+            .scrollableIfNecessary(enabled: self.configuration.mode.isFullScreen)
```
Comment (myeyesareblind) [inline RevenueCatUI/Templates/LinTemplate5View.swift:53]: if I have more then 2 offerings or a multi-line title, will 'purchase' button be pushed out of visible area?

Comment (maxkonovalov) [reply RevenueCatUI/Templates/LinTemplate5View.swift:53]: Currently, the primary content (buttons, title, and other texts) is attempted to always be visible by shrinking the top image. So while these elements fit into the window vertically, there should be no scrolling.

**Review by kaiengelhardt — APPROVED** [review]

File: RevenueCatUI/Data/Constants.swift [hunk 2]
```diff
@@ -51,6 +51,10 @@ enum Constants {
             return nil
         }
     }
+    
+    static func defaultContentWidth(_ idiom: UserInterfaceIdiom) -> CGFloat? {
```
[the remark this answers is not in the document]
Comment (maxkonovalov) [reply RevenueCatUI/Data/Constants.swift:55]: I followed an existing convention from the neighboring methods, where the label is omitted, too.

File: RevenueCatUI/Templates/LinTemplate5View.swift (hunk 1 above)
Comment (maxkonovalov) [reply RevenueCatUI/Templates/LinTemplate5View.swift]: @myeyesareblind I updated the UI layout to always keep the Subscribe button out of scroll

**Review by myeyesareblind — APPROVED** [review]

**Pull request merged**